docs/authoring.md

```diff
@@ -259,3 +224,116 @@ There are some template parameters that have a special meaning:
+
+
+## React
+
+[React](https://reactjs.org) is the render engine that we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatible.
+
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better error stack traces.
+* Provides better support for separating code into more manageable chunks/components.
+* The readability of the template is much better compared to Nunjucks syntax.
+* Better tool support for development.
+
+When writing React templates you decide whether to use CommonJS, ES5, or ES6 modules since everything is bundled together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
+There you can find information about how the renderer works or how we transpile your template files.
+
+Your React template always require `@asyncapi/generator-react-sdk` as a dependency. `@asyncapi/generator-react-sdk` is required to access the `File` component which is required as a root component for a file to be rendered. Furthermore it provides some common components to make your development easier, like `Text` or `Indent`.
+
+Let's consider a basic React template as the one below called `MyTemplate.js`:
+
+```js
+import { File, Text } from "@asyncapi/generator-react-sdk";
+
+export default function({ asyncapi, params, originalAsyncAPI }) {
+  return (
+    <File name="asyncapi.md">
+      <Text>Some text that should render as is</Text>
+    </File>
+  );
+}
+```
+
+The exported default function returns a `File` component as a root component which the [Generator](https://github.com/asyncapi/generator) uses to figure out what file should be generated. In our case we overwrite the default functionality of saving the file as `MyTemplate.js` but instead use the filename `asyncapi.md`. It is then specified that we should render `Some text that should render as is\n` within that file. Notice the `\n` character at the end, this is something that is automatically added after `Text` component. 
+
+For further information about components, props etc. see the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk)
+
+### Common assumptions
+
+1. All files within the template directory are rendered if their root component is of type `File` and the path is not part of the `nonRenderableFiles` configuration.
+1. New lines are automatically added after each `Text` component.
+
+### Debugging React template
+With React, it enables you to debug your templates. Below is a section of common IDE's and how to enable debugging. 
+
+#### VSCode
```

Comment:
The thought was that it could be introduced as [suggested](https://github.com/asyncapi/generator/pull/453#discussion_r536196752) by @magicmatatjahu. I am neutral about this, I can understand why it should be changed since we only provide one example.

Reply:
exactly, so once we have more, we can introduce structure, now it looks strange unless you have another PR coming already 😄

Reply:
yea I actually agree, I'll change it.